librz/include/rz_analysis.h

```diff
@@ -711,8 +716,15 @@ typedef struct rz_analysis_var_t {
 
 	// below members are just for caching, TODO: remove them and do it better
 	int argnum;
-} RzAnalysisVar;
 
+	struct {
+		enum {
+			RZ_ANALYSIS_VAR_ORIGIN_NONE = 0, ///< Variable was created from rizin
+			RZ_ANALYSIS_VAR_ORIGIN_DWARF, ///< Variable was created from DWARF information
+		} kind; ///< Kind of origin
+		RzBinDwarfLocation *DWARF_location; ///< Location description from DWARF
+	} origin; ///< Origin of the variable, i.e. DWARF, PDB, OMF
```

Comment:
I recommend to store "origin" as a pointer and as a structu with members "kind" and union "u", so that it can be easily extended in the future without breaking the API/ABI. @thestr4ng3r @wargio @ret2libc what do you think?

Reply:
I am ok with both ways. A pointer introduces additional indirection, but may help with memory size when most vars don't have an origin (thus null) and the union grows.

Reply:
I am ok with both ways. A pointer introduces additional indirection, but may help with memory size when most vars don't have an origin (thus null) and the union grows.